typescript/schemas/src/internal/types.ts

```diff
@@ -1,11 +1,4 @@
-export type FoxglovePrimitive =
-  | "string"
-  | "float64"
-  | "uint32"
-  | "boolean"
-  | "bytes"
-  | "time"
-  | "duration";
```

Comment:
after seeing the impact of this elsewhere, it feels like we went from having special-case logic to deal with the fact that time/duration were almost-but-not-quite-primitives, to now having special-case logic to deal with the fact that they are almost-but-not-quite-schemas. I think my takeaway would be that having them be a separate `{ type: "time" }` (i.e. not lumping them in with all other `nested` types) might have resulted in less special casing. But none of this really matters too much as long as the right results are generated, because this api is all internal anyway. (hence how we never have cleaned up the `type: { type: ...` wart